examples/cookbook/recipe_shared_memory.cpp

```diff
@@ -0,0 +1,120 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+
+#include "umpire/config.hpp"
+
+#include "umpire/Allocator.hpp"
+#include "umpire/ResourceManager.hpp"
+#include "umpire/Umpire.hpp"
+#include "umpire/resource/HostSharedMemoryResource.hpp"
+#include "umpire/util/MemoryResourceTraits.hpp"
+
+#include "mpi.h"
+
+#include <chrono>
+#include <iostream>
+#include <string>
+#include <thread>
+
+//
+// For debugging purposes, this program uses the number of command line
+// as a flag to indicate the mode it should run in.  The modes are:
+//
+// 1) If run with no argument (ac==1), it will run as an MPI program.
+// 2) If run with 1 argument (ac==2), it will run as the parent non-mpi program.
+// 3) If run with 2 arguments (ac==3), it will run as the child non-mpi program.
+//
+// This will allow someone to launch this program as a Parent and Child in two
+// separate debugger session windows (possible in gdb or vscode).  When running
+// in the debugger session windows, no MPI will be used and the debugger must
+// be used for synchronization (by setting breakpoints).
+//
+int main(
+  int
+  ac
+  ,
+  char**
+  av
+)
```

Comment:
This hurts my brain!

Reply:
oops.  that was left over from when I had #ifdefs between those items (which I wasn't compiling with MPI)..  I fix...

Reply:
This isn't fixed.

Reply:
Fixed.